### PR TITLE
simulation cleanup in preparation for real sensors

### LINKED ISSUE #977: Overo needs to come back for sim
Maybe without wacky name and factored into a better place.  sim_log.tll was nice.

### PR DESCRIPTION
**Pull request by mlyle**
Cleans up many small issues in the simulation target in preparation for using real sensors
- Realtime scheduler class support on Linux
- Better timebase for pios_delay (clock_gettime)
- Ability to run real sensors module if a sensor is present
- Restore ability to log to an output file-- through normal logging module -- fixes #977 
- Lots of restructuring and code organization improvements.

### CODE REVIEW

**Comment by mlyle** [comment]
This is now at a place where it could make sense to merge it.

File: flight/PiOS/posix/pios_delay.c [hunk 1]
```diff
@@ -33,17 +33,80 @@
...
+
+	tm *= numer;
+	tm /= denom;
+
+	t->tv_nsec = tm % 1000000000;
+	t->tv_sec  = tm / 1000000000;
+
+	return 0;
+}
+#endif
```
Comment (tracernz) [inline flight/PiOS/posix/pios_delay.c]: Can you comment these so they can be easily matched up to the corresponding if please?

File: flight/PiOS/posix/pios_fileout.c [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+	setbuf(file_dev->fh, NULL);
+
+	*file_id = (uintptr_t) file_dev;
+	
+	return 0;
+}
+
+
+void PIOS_FILEOUT_ChangeBaud(uintptr_t file_id, uint32_t baud)
```
Comment (tracernz) [inline flight/PiOS/posix/pios_fileout.c]: You don't need this function at all. PIOS_COM checks to see if the set_baud member is non-NULL before calling it.

File: flight/PiOS/posix/pios_sys.c [hunk 3]
```diff
@@ -127,6 +165,49 @@ void PIOS_SYS_Init(void)
...
+			.sched_priority = 50
+		};
+
+		rc = sched_setscheduler(0, SCHED_RR, &sch_p);
+
+		if (rc) {
+			perror("sched_setscheduler");
+			exit(1);
+		}
+#endif
```
Comment (tracernz) [inline flight/PiOS/posix/pios_sys.c:212]: #else #error might be nice?

File: flight/PiOS/posix/pios_fileout.c (hunk 2 above)
Comment (mlyle) [inline flight/PiOS/posix/pios_fileout.c]: OK.  Teaches me to just un-critically copy code from pios_tcp.

File: flight/PiOS/posix/pios_sys.c (hunk 3 above)
Comment (mlyle) [inline flight/PiOS/posix/pios_sys.c:212]: will push #else printf/exit here

File: flight/PiOS/posix/pios_delay.c (hunk 1 above)
Comment (tracernz) [inline flight/PiOS/posix/pios_delay.c]: Beep.

Comment (mlyle) [inline flight/PiOS/posix/pios_delay.c]: Woops did not see this.  rebased in and force pushed.  Thanks.

**Comment by mlyle** [comment]
Cool, thanks much :)

**Pull request merged**